Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -255,32 +255,48 @@ NWC24::ErrorCode NetKDRequestDevice::KDDownload(const u16 entry_index,
     return NWC24::WC24_ERR_SERVER;
   }
 
-  // Check if the filesize is smaller than the header size.
-  if (response->size() < sizeof(NWC24::WC24File))
+  if (m_dl_list.IsNotRSASigned(entry_index))
   {
-    ERROR_LOG_FMT(IOS_WC24, "File at {} is too small to be a valid file.", url);
-    LogError(ErrorType::KD_Download, NWC24::WC24_ERR_BROKEN);
-    return NWC24::WC24_ERR_BROKEN;
+    // Data that is not signed with an RSA key will not have the WC24 header or 320 bytes before the
+    // actual data. We just have to make sure that the response is not empty.
+    if (response->empty())
+    {
+      ERROR_LOG_FMT(IOS_WC24, "File at {} is empty.", url);
+      LogError(ErrorType::KD_Download, NWC24::WC24_ERR_BROKEN);
+      return NWC24::WC24_ERR_BROKEN;
+    }
+
+    file_data = *response;
   }
+  else
+  {
+    // Check if the filesize is smaller than the header size.
+    if (response->size() < sizeof(NWC24::WC24File))
+    {
+      ERROR_LOG_FMT(IOS_WC24, "File at {} is too small to be a valid file.", url);
+      LogError(ErrorType::KD_Download, NWC24::WC24_ERR_BROKEN);
+      return NWC24::WC24_ERR_BROKEN;
+    }
 
-  // Now we read the file
-  NWC24::WC24File wc24File;
-  std::memcpy(&wc24File, response->data(), sizeof(NWC24::WC24File));
+    // Now we read the file
+    NWC24::WC24File wc24File;
```

Comment:
I realize you just indented this but this is the wrong style, it should be `wc24_file`. Can you fix that while you're here?